### PR TITLE
Extract harvester registry

### PR DESCRIPTION
**Pull request by ruflin**
Extract harvester registry from log type to make it available to all prospectors. Add interface for harvesters.

### CODE REVIEW

**Comment by ruflin** [comment]
jenkins, retest it

**Comment by ruflin** [comment]
jenkins, retest it

File: filebeat/harvester/registry.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+// NewRegistry creates a new registry object
+func NewRegistry() *Registry {
+	return &Registry{
+		harvesters: map[uuid.UUID]Harvester{},
+	}
+}
+
+func (hr *Registry) add(h Harvester) {
```
Comment (7AC) [inline filebeat/harvester/registry.go]: nit: I'd keep the receiver to a single character (`r`)

Comment (ruflin) [inline filebeat/harvester/registry.go]: Will change that in the follow up PR.

**Pull request merged**